### DISCUSSION TITLE
Getting it going

### DISCUSSION THREAD

**Discussion by Tecnologic** [heinemannj/huawei_solar_hems#4] [General]
I think I have all the integrations setup according to the docs but my sensor.emhass_prediction_horizon is staying unknown. sun2 prediction_end has a value. What else is there what my prevent the prediction horizon from being calculated?

**Comment by Tecnologic** [comment]
for me all entities from huawei solar extension are called battery not batteries. I tried to fix this in all the files

**Reply by heinemannj** [reply]
You have to upgrade to the latest version of `Huawei Solar Integration` - Mandatory for HEMS!
You have to reconfigure `Huawei Solar Integration` from the scratch - Potentially delete the integration and to install again!

All obsolete `battery` entities have been renamed to `batteries` by `Huawei Solar Integration` - HEMS is using exclusively this actual naming convention - Mandatory for HEMS as well!

**Comment by Tecnologic** [comment]
By that i found this, what is this for? The heatpump?
            'sg_ready_evu': 'switch.406_wp_sg_ready_evu',
            'sg_ready_pv': 'switch.406_wp_sg_ready_pv_ma'

**Reply by heinemannj** [reply]
Yes - for the heat pump only - you can comment out if not required.

**Comment by heinemannj** [comment]
> I think I have all the integrations setup according to the docs but my sensor.emhass_prediction_horizon is staying unknown. sun2 prediction_end has a value. What else is there what my prevent the prediction horizon from being calculated?

Breaking changes for this kind of sensors during the last days.
YAML files has already updated, but the documentation is not up to date for the moment.
You have to wait for a couple of days until the parts of new introduced bluebrints (automations and scripts) have been explained.

And last but not least:
**Without providing error logs and sensor outputs I can't help you in detail!**

**Reply by Tecnologic** [reply]
Thanks U helped a lot. Looks like I used a fork of Huawei solar integration. I will switch to the original version and start my setup from scratch again to have no old entities. When I'm clean on that part I come back to the docs.

May be a quick remark for the docs. Some early sanity check may be nice. Like does emhass day ahead prediction work, is entity X available. 

And another "improvement" would be a customizing section. I have a different Huawei inverter and battery Ande there are a lot of places where the powers and battery sizes are magic numbers. Maybe a central yaml section setting these valves may be nice.

All in this is a great project and I'm eager to get it working right. If U need help or a tester just point me to it.

**Comment by heinemannj** [comment]
Try this `Auto Entities Card` configuration to check all sensor states used by HEMS:

```
type: custom:auto-entities
card:
  type: entities
filter:
  template: |
    {% from '032-hems.jinja' import globalVariables %}
    {%  set var = globalVariables()|from_json %}
    
    {% for name in var|list %}
    {{ var[name] }}
    {% endfor %}
```

**You have to use the latest HEMS updates from this afternoon!!!**

**Reply by Tecnologic** [reply]
Thanks. I'll try it tomorrow. Which version of Huawei solar are U referring to? 1.4.1 or the 1.5 alpha?

**Comment by heinemannj** [comment]
I'm using the latest alpha, but the latest stable 1.4.x version is also fine.

**Reply by Tecnologic** [reply]
I'm on the alpha now too. A delete and reconnect changed all the battery entities to batteries. Will do the same for the other integrations tonight.

**Reply by Tecnologic** [reply]
but my sensors are now like this: `sensor.batteries_batterieladung` not `sensor.batteries_state_of_capacitance`

**Comment by Tecnologic** [comment]
https://github.com/heinemannj/huawei_solar_hems/wiki/EMHASS-Add%E2%80%90on#electricity-spot-price-forecast

How do i set the entity and setting columns of this table?

**Reply by Tecnologic** [reply]
These values are comming from the rest request of hems to emhass?

**Comment by Tecnologic** [comment]
With huawei_solar 1.50 alpha which generated localized entities, isn't there a portable way of accessing the data of the extension.

Right now i'm fixing all the references in the hems code. By that i saw that 1.50 change the battery charge/discharge power entities to a combined entity. `sensor.batteries_lade_entladeleistung` in my setup.  In the package file for huawei solar card
`  #
  # Average of Battery Discharge Power (W) in last 24h
  #
  - platform: statistics
    name: batteries_discharge_power_mean_24h
    unique_id: batteries_discharge_power_mean_24h
    entity_id: sensor.batteries_lade_entladeleistung
    state_characteristic: mean
    max_age:
      hours: 24
  #
  # Average of Battery Charge Power (W) in last 30mins
  #
  - platform: statistics
    name: batteries_charge_power_mean_30min
    unique_id: batteries_charge_power_mean_30min
    entity_id: sensor.batteries_lade_entladeleistung
    state_characteristic: mean
    max_age:
      hours: 0.5
#`
 here the sign of the average discharge power would change
